MaxText/layers/linears.py

```diff
@@ -629,6 +666,33 @@ def dense_matmul(self, inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel):
         )
       return output, None
 
+  def retrieve_quantized_weight(self, inputs, gate_logits) -> tuple[QTensor, QTensor, QTensor]:
+    # This is called only during tracing. This is to invoke creation of quantized tensor inside AqtEinsum.
+    # After jit, this will become no-op and will not affect performance.
+    _ = self.capped_dense(inputs, gate_logits)
```

Comment:
I am not quite understand this logic, but it seems we have to do so :)

Reply:
AqtEinsum is a flax compact class which initializes variable only when einsum is called. 
So to invoke creation of quantized variable in `aqt` collection, we manually invoke AqtEinsum and read parameters using `aqt` collection. This is kind of unaovidable with current deign of AqtEinsum unfortunately.